store.go

```diff
@@ -69,6 +70,67 @@ func (store *Store) Walk(ctx context.Context, prefix string, walk func(key strin
 	return
 }
 
+// KeyData is a representation of a key in Consul, which follows the structure
+// documented at https://www.consul.io/api/kv.html#read-key
+type KeyData struct {
+	CreateIndex int64
+	ModifyIndex int64
+	Key         string
+	Flags       int64
+	Value       []byte
+}
+
+// WalkData traverses the keyspace under the given prefix, calling the walk
+// function and passing each key's data
+//
+// If the walk function returns an error the iteration is stopped and the error
+// is returned by WalkData
+func (store *Store) WalkData(ctx context.Context, prefix string, walk func(data KeyData) error) (err error) {
+	var result io.ReadCloser
+	var query = Query{
+		{Name: "recurse", Value: "true"},
+	}
+
+	if _, result, err = store.client().do(ctx, "GET", store.path(prefix), query, nil); err != nil {
+		return
+	}
+	defer result.Close()
+
+	var stream = json.NewStreamDecoder(result)
+
+	// What returns from the HTTP response isn't quite what is needed, so start
+	// here as an intermediate and then build a KeyData from the raw repr
+	var rawData struct {
```

Comment:
Why not use `KeyData` directly? The JSON decoder will parse the base64 encoded value.